src/reaper_osara.cpp

```diff
@@ -1298,7 +1302,19 @@ LRESULT CALLBACK keyboardHookProc(int code, WPARAM wParam, LPARAM lParam) {
 	} else if (wParam == VK_F6 && !(lParam & 0x80000000)) {
 		if (maybeSwitchToFxPluginWindow())
 			return 1;
-	}
+	}else if(wParam == VK_HOME || wParam == VK_END || wParam == VK_SHIFT){
+		//Pressing home / end keys in midi events view moves the cursor to top / bottom respectively, by passing the needed key events to the listview
+		if(wcscmp(className, L"SysListView32") == 0){
+			HWND parent = GetParent(focus);
+			WCHAR parentClassName[22] = L"\0";
+			GetClassNameW(parent, parentClassName, ARRAYSIZE(parentClassName));
+			if(wcscmp(parentClassName, L"REAPERmidieditorwnd") == 0){
+				int msg = WM_KEYDOWN;
+				if((lParam >> 31) == 1) msg=WM_KEYUP;
+				SendMessage(focus, msg, wParam, 0);
```

Comment:
So we're just sending the key message to the ListView? It'd be good if you can verify whether the home/end keys work in the events list without OSARA. I'm concerned that OSARA is actually the problem here and is preventing those keys from getting through. Otherwise, I guess REAPER's own key hook is eating home/end? Sounds like something that should ideally be fixed in REAPER. The Cockos devs are usually pretty responsive to reports like these.

Reply:
I re-tested the following.
1) Checkout the latest master 94f67edff6081d5e1f892f4f969b5b5ae15c5886
2) build
3) home / end do not work
If further tests are needed, please let me know. I'm still not sure if this test is enough.
I don't know what is eating the key event, but something is definitely doing so.
I guess it is one of the parents at least, since directly sending the key event to the listview itself changes the cursor position.
Are we going to continue working on this PR? Or should we report this issue to Reaper devs? What do you think about this?

Reply:
Yeah, I guess that would indeed be tough. Perhaps just comment out the SetWindowsHookEx for the keyboard hook. I don't see how anything else in OSARA could be interfering here. You won't be able to use the applications key with that hook disabled, but you should be able to get into the MIDI editor with control+alt+e (I think; it's something like that).

Reply:
Sure, thanks. I will post another comment after testing is done. Then, let's decide whether we report it to Cockos or not.